chat-plugins/dexsearch.js

```diff
@@ -11,13 +11,153 @@
+					this.parse(message.parse);
+				}
+			});
+		},
+		randompokemonhelp: ["/randompokemon - Generates random Pok\u00e9mon based on given search conditions.",
+			"/randompokemon uses the same parameters as /dexsearch (see '/help ds').",
+			"Adding a number as a parameter returns that many random Pok\u00e9mon, e.g., '/randpoke 6' returns 6 random Pok\u00e9mon."],
+
+		ms: 'movesearch',
+		msearch: 'movesearch',
+		movesearch: function (target, room, user, connection, cmd, message) {
+			if (!this.canBroadcast()) return;
+
+			if (!target) return this.parse('/help movesearch');
+
+			queryChild({
+				target: target,
+				cmd: 'movesearch',
+				canAll: (!this.broadcasting || room.isPersonal),
+				message: (this.broadcasting ? "" : message),
+			}).then(message => {
+				if (message.reply) {
+					this.sendReplyBox(message.reply);
+				} else if (message.parse) {
+					this.parse(message.parse);
+				}
+			});
+		},
+		movesearchhelp: ["/movesearch [parameter], [parameter], [parameter], ... - Searches for moves that fulfill the selected criteria.",
+			"Search categories are: type, category, flag, status inflicted, type boosted, and numeric range for base power, pp, and accuracy.",
+			"Types must be followed by ' type', e.g., 'dragon type'.",
+			"Stat boosts must be preceded with 'boosts ', e.g., 'boosts attack' searches for moves that boost the attack stat.",
+			"Inequality ranges use the characters '>' and '<' though they behave as '≥' and '≤', e.g., 'bp > 100' searches for all moves equal to and greater than 100 base power.",
+			"Parameters can be excluded through the use of '!', e.g., !water type' excludes all water type moves.",
+			"Valid flags are: authentic (bypasses substitute), bite, bullet, contact, defrost, powder, pulse, punch, secondary, snatch, sound",
+			"If a Pok\u00e9mon is included as a parameter, moves will be searched from it's movepool.",
+			"The order of the parameters does not matter."],
+
+		isearch: 'itemsearch',
+		itemsearch: function (target, room, user, connection, cmd, message) {
+			if (!target) return this.parse('/help itemsearch');
+			if (!this.canBroadcast()) return;
+
+			queryChild({
+				target: target,
+				cmd: 'itemsearch',
+				canAll: (!this.broadcasting || room.isPersonal),
+				message: (this.broadcasting ? "" : message),
+			}).then(message => {
+				if (message.reply) {
+					this.sendReplyBox(message.reply);
+				} else if (message.parse) {
+					this.parse(message.parse);
+				}
+			});
+		},
+		itemsearchhelp: ["/itemsearch [move description] - finds items that match the given key words.",
+		"Command accepts natural language. (tip: fewer words tend to work better)",
+		"Searches with \"fling\" in them will find items with the specified Fling behavior.",
+		"Searches with \"natural gift\" in them will find items with the specified Natural Gift behavior."],
+	};
+} else {
+	// This is the child
+
+	global.Tools = require('../tools.js');
+	global.toId = Tools.getId;
+	Tools.includeData();
```

Comment:
Fortunately for you, I just recently updated it so `TeamValidator` doesn't spawn a child process unless you force it to! How convenient!